---

title: peptide


keywords: fastai
sidebar: home_sidebar

summary: "An open source library for peptide classification using Machine Learning and Deep Learning."
description: "An open source library for peptide classification using Machine Learning and Deep Learning."
nb_path: "nbs/index.ipynb"
---
<!--

#################################################
### THIS FILE WAS AUTOGENERATED! DO NOT EDIT! ###
#################################################
# file to edit: nbs/index.ipynb
# command to build the docs after a change: nbdev_build_docs

-->

<div class="container" id="notebook-container">
        
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>This project aims to build a series of classifiers that can predict whether a given amino acid sequence is one or more of the 3 target peptides.<br>
Additionally, this project will compare models built with vastly different approaches ranging from classic ML models with feature engineering by experts to large transformer-based models.</p>
<p><strong>Classification Tasks:</strong> Given a sequence of amino acids (NLP equivalent = ‘words’), classify whether the resulting peptide (NLP equivalent = ‘sentence’) is one or many of the following targets (multi-task classification not multi-class i.e. 3 sigmoids not 1 softmax).</p>
<ul>
<li>Anticancer peptide (ACP)</li>
<li>DNA-binding protein</li>
<li>Antimicrobial peptide (AMP).</li>
</ul>
<p><strong>Multiple Models:</strong> Develop multiple models for the above classification tasks</p>
<ul>
<li>Supervised ML model(s) with hand engineered features from bio-experts</li>
<li>Supervised ML model(s) using pre-trained embeddings with feature-space reduced using the following unsupervised techniques<ul>
<li>Principal Component Analysis (PCA)</li>
<li>Autoencoders &amp; its variants</li>
<li>K-Means Clustering</li>
</ul>
</li>
<li>Deep Learning Transformer models using pre-trained embeddings</li>
</ul>
<p><strong>Model Scoring System:</strong> Develop a system that produces accuracy scores</p>
<ul>
<li>By running any given (labeled) dataset</li>
<li>Through the multiple models listed above </li>
<li>And display / return accuracies on the classification tasks for each model.</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Install">Install<a class="anchor-link" href="#Install"> </a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>With conda</p>
<ul>
<li><code>conda install -c conda-forge peptide</code></li>
</ul>
<p>With pip</p>
<ul>
<li><code>pip install peptide</code></li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="How-to-use">How to use<a class="anchor-link" href="#How-to-use"> </a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Fill me in please! Don't forget code examples:</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="mi">1</span><span class="o">+</span><span class="mi">1</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>2</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

</div>


